cloudinit/cmd/status.py

```diff
@@ -16,11 +17,15 @@
 CLOUDINIT_DISABLED_FILE = "/etc/cloud/cloud-init.disabled"
 
 # customer visible status messages
-STATUS_ENABLED_NOT_RUN = "not run"
-STATUS_RUNNING = "running"
-STATUS_DONE = "done"
-STATUS_ERROR = "error"
-STATUS_DISABLED = "disabled"
+@enum.unique
+class UXAppStatus(enum.Enum):
+    """Enum representing user-visible cloud-init application status."""
+
+    ENABLED_NOT_RUN = "not-run"
```

Comment:
Idea: Besides `"not-run"` the enum names are just the lower-case equivalent of the enum name. If they all matched, I would suggest making the return codes the enum value to store the rc alongside the return message (and then access the enum via `enum.name` or `enum.name.lower()` when needed for printing messages).

Reply:
I'm probably not understanding the intent of your comment here. If get_status_details returned the lower enum value to the caller `handle_status_args` wouldn't I then have to [decode those via UXAppStatus.name.lower() here too](https://github.com/canonical/cloud-init/pull/1162/files#diff-da437f8452f6c63679a5c5e6abb6ac5e999c9bf7ca6717a3e026de77da6b6608R71)

Reply:
You would, yes. The point I'm making is that we define an enum data structure that contains strings and later use the value of the enum to derive error codes [here](https://github.com/canonical/cloud-init/pull/1162/files#diff-df0c5bb55245c37f1e8cd155d17f3e29eaaf2f82bb7357445803e44e4a27c305R71).

This suggestion was intended to be "rather than derive error codes programmatically why not to store them together with their intended string representation, since it appears that we have a 1:1 mapping". The specifics I mentioned with `.lower()` is one way to do that given the current `enum` usage. Also this suggestion might be unnecessary for a data structure that isn't widely used - I'm fine with it as is if you prefer it the way it was.

Reply:
This is a really good pt @holman. I thought I would adopt this approach as suggested, but forgot that RUNNING vs DONE status types would have the common non-unique exit code of 0. There may be other enum status that grow over time that also might fall into exit-code 1 for general failure. So, I think I won't be able to use enum to represent this structure.